src/session.ts

```diff
@@ -80,38 +88,43 @@ export class Session extends Client implements ErdstallSession {
 		return this.getAccount(this.address);
 	}
 
-	async transferTo(assets: Assets, to: Address): Promise<TxReceipt> {
+	async transferTo(assets: Assets, to: Address): Promise<PendingTransaction> {
 		if (!this.erdstallConn) {
-			return Promise.reject(ErrUnitialisedClient);
+			throw ErrUnitialisedClient;
 		}
 		const nonce = await this.nextNonce();
 		const tx = new Transfer(this.address, nonce, to, assets);
-		await tx.sign(this.erdstallConn.erdstall(), this.signer);
-		return this.enclaveWriter.transfer(tx);
+		await tx.sign(this.erdstallConn!.erdstall(), this.signer);
+		const hash = tx.hash(this.erdstallConn!.erdstall());
+		const receipt = this.receiptDispatcher.register(hash);
+		const accepted = this.enclaveWriter.transfer(tx);
+		return { receipt, accepted };
 	}
 
-	async mint(token: Address, id: Uint256): Promise<TxReceipt> {
+	async mint(token: Address, id: Uint256): Promise<PendingTransaction> {
 		if (!this.erdstallConn) {
-			return Promise.reject(ErrUnitialisedClient);
+			throw ErrUnitialisedClient;
 		}
-
-		const minttx = new Mint(
-			this.address,
-			await this.nextNonce(),
-			token,
-			id,
-		);
-		await minttx.sign(this.erdstallConn.erdstall(), this.signer);
-		return this.enclaveWriter.mint(minttx);
+		const nonce = await this.nextNonce();
+		const tx = new Mint(this.address, nonce, token, id);
+		await tx.sign(this.erdstallConn!.erdstall(), this.signer);
+		const hash = tx.hash(this.erdstallConn!.erdstall());
+		const receipt = this.receiptDispatcher.register(hash);
+		const accepted = this.enclaveWriter.mint(tx);
+		return { receipt, accepted };
```

Comment:
Last comment, rest is fine: Could we create a small helper method so we do not duplicate this code in every TX function?

Reply:
Keep it like it is because it is simple and the alternative contains black magic JS regarding the scoping of `this` and might even be more cumbersome to use.